### PR TITLE
Add issue-wanted idea

### PR DESCRIPTION
**Pull request by ChShersh**

### CODE REVIEW

**Comment by ChShersh** [comment]
> I guess this is important and motivating info, what do you think?

Yes, I think this is a good idea :+1: Will add these sections as well.

File: content/ideas/issue-wanted.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+3. Elm frontend that displays this information.
+
+This project is a great way for students to work on real-world Haskell project
+using modern libraries and approaches.
+
+**Potential Mentors**: Dmitrii Kovanikov, Veronika Romashkina
+
+**Difficulty**: Intermediate
+
+Mentors take care of Elm frontend and deployment. Student can work only on
```
[the remark this answers is not in the document]
Comment (ChShersh) [reply content/ideas/issue-wanted.md]: I wanted to say that if student want, he/she also can work on frontend and DevOps tasks :slightly_smiling_face: Saying that we can take care of this seems good enough to me :+1:

File: content/ideas/issue-wanted.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+Haskell part. The choice of main libraries for the project is the following:
```
[the remark this answers is not in the document]
Comment (ChShersh) [reply content/ideas/issue-wanted.md]: I think it's better to say `suggested` as you proposed :+1: Though, I really don't want to use `spock` or something like that. And in that case I can't really mentor the project.

File: content/ideas/issue-wanted.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+* `servant` for writing REST API servers and clients in Haskell
+* PostgreSQL as database using `postgresql-simple` Haskell library
+* JSON as a communication format between backend and frontend, `aeson` Haskell library
+* `async` for writing concurrent code
```
[the remark this answers is not in the document]
Comment (ChShersh) [reply content/ideas/issue-wanted.md]: Not sure... We haven't write this library yet :slightly_smiling_face:

File: content/ideas/issue-wanted.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
+---
+title: "issue-wanted: Web app for discovering potential contributions to the Haskell ecosystem"
+---
+
+Haskell ecosystem can be improved in many ways. There many libraries of different categories
```
Comment (vrom911) [inline content/ideas/issue-wanted.md]:
```suggestion
Haskell ecosystem can be improved in many ways. There are many libraries of different categories
```

File: content/ideas/issue-wanted.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+with tons of issues — starting with low-hanging fruits to expert level problems. However, there is no easy way to discover potential problems you
+would like to work on. The goal of [issue-wanted](https://github.com/kowainik/issue-wanted)
```
Comment (vrom911) [inline content/ideas/issue-wanted.md]: Work on or work at? :thinking:

File: content/ideas/issue-wanted.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+is to help find GitHub issues using GitHub labels and various metadata fields
```
Comment (vrom911) [inline content/ideas/issue-wanted.md]: "find project issues"?

File: content/ideas/issue-wanted.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+The project contains several parts:
+
+1. Haskell process that syncs information about Haskell repositories at GitHub.
+2. Haskell backend that provides REST API for fetched information.
+3. Elm frontend that displays this information.
+
+The benefits of this project are the following:
+
+1. Easier for beginners and experts to find issues they can solve.
```
Comment (vrom911) [inline content/ideas/issue-wanted.md]: With `issue-wanted` it's more accessible fo beginners to find interesting issues they can solve.

File: content/ideas/issue-wanted.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+2. Average maintenance of Haskell libraries will be improved as consequence: if
```
Comment (vrom911) [inline content/ideas/issue-wanted.md]:
```suggestion
2. Average maintenance of Haskell libraries will be improved as a consequence: if
```

File: content/ideas/issue-wanted.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+   people want more open-source contributors, they will provide better meta
```
Comment (vrom911) [inline content/ideas/issue-wanted.md]: probably "project owners" instead of "people"?

Comment (vrom911) [inline content/ideas/issue-wanted.md]: They *could* provide..

File: content/ideas/issue-wanted.md [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+   information to their packages and this actuall will pay off.
+
```
Comment (vrom911) [inline content/ideas/issue-wanted.md]: typo in `actually`, also better not to use *will*, you can say this pays off (?)

I would add that any project could become more popular and gather more attention, but not sure how to tell this correctly..

File: content/ideas/issue-wanted.md [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+1. Haskell process that syncs information about Haskell repositories at GitHub.
+2. Haskell backend that provides REST API for fetched information.
+3. Elm frontend that displays this information.
+
+The benefits of this project are the following:
+
+1. Easier for beginners and experts to find issues they can solve.
+2. Average maintenance of Haskell libraries will be improved as consequence: if
+   people want more open-source contributors, they will provide better meta
+   information to their packages and this actuall will pay off.
```
Comment (vrom911) [inline content/ideas/issue-wanted.md]:
```suggestion
   information to their packages, and this actuall will pay off.
```

File: content/ideas/issue-wanted.md [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+
+This project is an excellent way for students to work on real-world Haskell project
```
Comment (vrom911) [inline content/ideas/issue-wanted.md]:
```suggestion
This project is an excellent way for students to work on the real-world Haskell project
```

File: content/ideas/issue-wanted.md [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+using modern libraries and approaches.
+
+**What is expected from student**
+
+1. Basic knowledge of Haskell programming language: `newtype`, typeclasses,
+   `Monad`s, ability to write code with `Reader` or `State` monad.
+2. Very basic knowledge of SQL. At least student should not be afraid to learn
+   and use SQL, though it's not expected to have complicated queires in this
```
Comment (vrom911) [inline content/ideas/issue-wanted.md]:
```suggestion
   and use SQL, though it's not expected to have any complex queries in this
```

File: content/ideas/issue-wanted.md [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+2. Average maintenance of Haskell libraries will be improved as consequence: if
+   people want more open-source contributors, they will provide better meta
+   information to their packages and this actuall will pay off.
+
+This project is an excellent way for students to work on real-world Haskell project
+using modern libraries and approaches.
+
+**What is expected from student**
+
+1. Basic knowledge of Haskell programming language: `newtype`, typeclasses,
```
Comment (vrom911) [inline content/ideas/issue-wanted.md]: knowledge of Haskell programming language *concepts*/*definitions*/*something else*?

File: content/ideas/issue-wanted.md [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+   `Monad`s, ability to write code with `Reader` or `State` monad.
+2. Very basic knowledge of SQL. At least student should not be afraid to learn
+   and use SQL, though it's not expected to have complicated queires in this
+   project.
+3. Patience. Reading GitHub API documentation and testing integration of
+   multiple systems might be frustrating sometimes.
+
+**What student will get after this project**
```
Comment (vrom911) [inline content/ideas/issue-wanted.md]: Will or would?

File: content/ideas/issue-wanted.md [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+
+1. Experience of writing REST API web services in Haskell.
+2. Familarity with intermediate Haskell features like monad transformers and
```
Comment (vrom911) [inline content/ideas/issue-wanted.md]:
```suggestion
2. Familiarity with intermediate Haskell features like monad transformers and
```

File: content/ideas/issue-wanted.md [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+   type-level computations.
+3. Knowledge of modern Haskell programming patterns and idioms.
+
+**Potential Mentors***: Dmitrii Kovanikov, Veronika Romashkina
+
+**Difficulty**: Intermediate
+
+Mentors can take care of Elm frontend and deployment. The student can work only
+on the Haskell part (but also can work on frontend and DevOps tasks). Suggested
```
Comment (vrom911) [inline content/ideas/issue-wanted.md]: Probably better to say, that it's not restricted to Haskell only, if they have any interest in other details of the projects it's free to experiment with. Or something in better word, I hope you got the idea I wanted to point out to

File: content/ideas/issue-wanted.md (hunk 5 above)
Comment (ChShersh) [reply content/ideas/issue-wanted.md]: Looks like they both equivalent:

* https://ell.stackexchange.com/a/98307

But for me `work at` usually means some place, like `work at Kowainik`

**Pull request merged**